File: superhero-api/src/main/java/com/viceri/herois/superhero_api/model/Hero.java
package com.viceri.herois.superhero_api.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Hero {
    private Long id;
    private String nome;
    private String nomeHeroi;
    private LocalDate dataNascimento;
    private Double altura;
    private Double peso;
    private List<Superpower> superpoderes;
}

File: frontend/superhero-client/src/app/core/models/hero.ts
import { Superpower } from "./super_power";


export interface Hero {
  id: number;
  nome: string;
  nomeHeroi: string;
  dataNascimento: string; 
  altura: number;
  peso: number;
  superpoderes: Superpower[];
}

export interface HeroRequest {
  nome: string;
  nomeHeroi: string;
  dataNascimento: string; 
  altura: number;
  peso: number;
  superpoderesIds: number[];
}